packages/astro/src/core/dev/index.ts

```diff
@@ -321,6 +320,7 @@ export class AstroDevServer {
         routeCache: this.routeCache,
         viteServer: this.viteServer,
       });
+      this.mostRecentRoute = route;
```

Comment:
This was just moved down below the `ssr()` call so it doesn’t get cached on error